deform/tests/test_schema.py

```diff
@@ -20,6 +20,20 @@ def _makeOne(self):
 
         return FileData()
 
+    def test_deferred_csrf_value(self):
+        import colander
+        from deform.schema import CSRFSchema
+        from deform.schema import deferred_csrf_value
+        new_schema = CSRFSchema()
+        from unittest.mock import MagicMock
+        kw = {"request": colander.null}
+        kw["request"].session = MagicMock(name="session")
+        kw["request"].session.get_csrf_token = MagicMock(
+            return_value=colander.null)
```

Comment:
That's true, we could return other value than colander.null, and remove the import colander, however collander.null is what being returned when function is decorated with @colander.deferred, until later on when the schema is bound to request, and that is the time kw['request'].session.get_csrf_token(node, kw) being accessed.
 
The way it is now is the closest to decorated function.

I wait for more feedbacks, but previous developers omitted schema.py and renderer.py for this reason that request object can't be instantiated easily outside of view, we could create a dummy version of request, but that would be the same as mocking it :)

Reply:
Thanks for the explanation. I think your approach is good. I approved this PR.

I could merge this PR now or wait a day for additional comments. Either way, folks can always add comments. Please let me know. Thank you!